# hunk 1: multi-inheritance is broken https://sourceforge.net/p/swig/bugs/1379/
# hunk 2: thread bug on iterators https://sourceforge.net/p/swig/bugs/1370/
# hunk 3: compilation issue on GCC 4.6 http://sourceforge.net/p/swig/patches/296/
--- eossdk_wrap.cpp-t
+++ eossdk_wrap.cpp-t
@@ -1618,6 +1618,10 @@
   if (!SwigPyObject_Check(next)) {
     return NULL;
   }
+  while (sobj->next && SwigPyObject_Check(sobj->next)) {
+     sobj = (SwigPyObject *)sobj->next;
+  }
+  ((SwigPyObject *)next)->next = sobj->next;
   sobj->next = next;
   Py_INCREF(next);
   return SWIG_Py_Void();
@@ -3495,7 +3499,11 @@
     }
       
   public:
-    virtual ~SwigPyIterator() {}
+    virtual ~SwigPyIterator() {
+       SWIG_PYTHON_THREAD_BEGIN_BLOCK;
+       _seq = 0;
+       SWIG_PYTHON_THREAD_END_BLOCK;
+    }
 
     // Access iterator method, required by Python
     virtual PyObject *value() const = 0;
@@ -3801,6 +3809,7 @@
 
 
 #include <stdexcept>
+#include <cstddef>
 
 
 #include <algorithm>
